api-admin/src/main/java/bio/terra/pearl/api/admin/service/DataRepoExportExtService.java

```diff
@@ -27,12 +29,12 @@ public DataRepoExportExtService(
     this.studyEnvironmentService = studyEnvironmentService;
   }
 
+  @EnforcePortalPermission(permission = "tdr_export")
   public List<Dataset> listDatasetsForStudyEnvironment(
       String portalShortcode,
       String studyShortcode,
       EnvironmentName environmentName,
       AdminUser user) {
-    authUtilService.authUserToPortalWithPermission(user, portalShortcode, "tdr_export");
     authUtilService.authUserToStudy(user, portalShortcode, studyShortcode);
```

Comment:
I do really like this syntax, but I feel like we often use these auth methods like:

```
Portal portal = authUtilService.authUserToPortal<WithPermission>(user,...)
.... use portal somehow ....
```

Because of that, I'm not 100% in love that we would then have two places where authentication could happen (direct authUtilService calls and annotations). If this is a problem, we could just go full annotations for authentication, which would then maybe mean a couple more db calls to grab the portal/study in the cases we need it. But maybe it's not a problem and it's ok to do in two places!

Reply:
that's a great point -- I'll hold off on merging this for now, and take a look at how an annotation might also populate an "PortalAuthContext" object that the method could access, because you're right that there are lots of cases where we want that Portal object, and it's very important that that portal object is the same one we authed to.